pkg/controller/pravega/pravega_segmentstore.go

```diff
@@ -389,12 +389,12 @@ func MakeSegmentStoreExternalServices(pravegaCluster *api.PravegaCluster) []*cor
 
 	serviceType := getSSServiceType(pravegaCluster)
 	services := make([]*corev1.Service, pravegaCluster.Spec.Pravega.SegmentStoreReplicas)
-	var annotationMap map[string]string
 
 	for i := int32(0); i < pravegaCluster.Spec.Pravega.SegmentStoreReplicas; i++ {
-		annotationMap = map[string]string{}
+		annotationMap := map[string]string{}
 		ssPodName := util.ServiceNameForSegmentStore(pravegaCluster.Name, i)
 		annotationValue := generateDNSAnnotationForSvc(pravegaCluster.Spec.ExternalAccess.DomainName, ssPodName)
+		annotationMap = pravegaCluster.Spec.Pravega.SegmentStoreServiceAnnotations
```

Comment:
Please add unit tests to cover the usecase where domainName is not specified but serviceAnnotations are.

Reply:
done